src/storage/src/source/postgres/replication.rs

```diff
@@ -652,18 +664,24 @@ async fn raw_stream<'a>(
         slot_metadata.active_pid, wal_sender_timeout
     );
 
-    let progress_stat_shared_value = Arc::new(Mutex::new(None));
-    let progress_stat_task_value = Arc::clone(&progress_stat_shared_value);
+    let (probe_tx, mut probe_rx) = watch::channel(None);
     let offset_known_interval =
         mz_storage_types::dyncfgs::PG_OFFSET_KNOWN_INTERVAL.get(config.config.config_set());
+    let now_fn = config.now_fn.clone();
     let max_lsn_task_handle =
         mz_ore::task::spawn(|| format!("pg_current_wal_lsn:{}", config.id), async move {
             let mut interval = tokio::time::interval(offset_known_interval);
 
-            loop {
+            while !probe_tx.is_closed() {
                 interval.tick().await;
-                let lsn_or_err = super::fetch_max_lsn(&metadata_client).await;
-                *progress_stat_task_value.lock().expect("poisoned") = Some(lsn_or_err);
+                let probe_ts = mz_repr::Timestamp::try_from((now_fn)()).expect("must fit");
```

Comment:
I switched to a tokio Watch channel instead of a mutex since that also provides await based notifications which allows us to `select!` it below